esphome/components/sim800l/sim800l.h

```diff
@@ -45,29 +41,36 @@ class Sim800LComponent : public uart::UARTDevice, public PollingComponent {
   void add_on_sms_received_callback(std::function<void(std::string, std::string)> callback) {
     this->callback_.add(std::move(callback));
   }
+  void add_on_connected_callback(std::function<void()> callback) { this->callback_reg.add(std::move(callback)); }
   void send_sms(const std::string &recipient, const std::string &message);
+  void delete_sms();
   void dial(const std::string &recipient);
 
+  void set_rssi_(sensor::Sensor *rssi_);
+
  protected:
   void send_cmd_(const std::string &);
   void parse_cmd_(std::string);
 
   std::string sender_;
+  std::string message_;
   char read_buffer_[SIM800L_READ_BUFFER_LENGTH];
   size_t read_pos_{0};
   uint8_t parse_index_{0};
   uint8_t watch_dog_{0};
   bool expect_ack_{false};
   sim800l::State state_{STATE_IDLE};
   bool registered_{false};
-  int rssi_{0};
 
   std::string recipient_;
   std::string outgoing_message_;
   bool send_pending_;
   bool dial_pending_;
 
   CallbackManager<void(std::string, std::string)> callback_;
+  CallbackManager<void()> callback_reg;
```

Comment:
Protected variables need to end with `_`

```suggestion
  CallbackManager<void()> callback_reg_;
```